File: src/main/java/life/magpies/registry/ModRegister.java
package life.magpies.registry;

import life.magpies.registry.commands.ModCommands;
import life.magpies.registry.sounds.ModSounds;
import net.fabricmc.fabric.api.biome.v1.BiomeModifications;
import net.fabricmc.fabric.api.biome.v1.BiomeSelectors;
import net.minecraft.world.gen.GenerationStep;

import static life.magpies.registry.ModItems.CITRINE_ORE_PLACED_KEY;

public class ModRegister {
    public static void registerMod() {
        ModBlocks.registerBlock();
        ModItems.registerModItems();
        ModSounds.registerSounds();
        ModCommands.registerCommands();

        BiomeModifications.addFeature(BiomeSelectors.foundInOverworld(), GenerationStep.Feature.UNDERGROUND_ORES, CITRINE_ORE_PLACED_KEY);
    }
}

File: src/main/kotlin/life/magpies/registry/ModRegister.kt
package life.magpies.registry

import life.magpies.registry.ModItems.CITRINE_ORE_PLACED_KEY
import life.magpies.registry.commands.ModCommands
import life.magpies.registry.sounds.ModSounds
import net.fabricmc.fabric.api.biome.v1.BiomeModifications
import net.fabricmc.fabric.api.biome.v1.BiomeSelectors
import net.minecraft.world.gen.GenerationStep

object ModRegister {
    fun register() {
        ModItemGroups.buildItemGroups()
        ModItems.registerItemLog()
        ModBlocks.registerBlock()
        ModSounds.registerSoundsLog()
        ModCommands.registerCommands()
        BiomeModifications.addFeature(BiomeSelectors.foundInOverworld(), GenerationStep.Feature.UNDERGROUND_ORES, CITRINE_ORE_PLACED_KEY)
    }
}